packages/embed_api/src/createlyEmbed.ts

```diff
@@ -0,0 +1,187 @@
+import {PostMessageAPI} from './message-api';
+import { of, Observable, BehaviorSubject } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { throwError } from 'rxjs';
+
+interface Doctype {
+    iFrameId:string;
+    token:string;
+    docID:string;
+    docMode:string;
+}
+
+/**
+ * Defines the types of postmesssage send events
+ */
+export enum PostMessageRecvEventType {
+    /**
+     * Fired when a shape is touched
+     */
+    shapeTouch = 'shape:touch',
+
+
+    /**
+     * Fired when the selection changes
+     */
+    shapeSelect = 'shape:select',
+
+
+    /**
+     * Fired when the document is loaded
+     */
+    documentLoad = 'document:load',
+
+
+
+    /**
+     * Fired when the document is loaded
+     */
+    documentReady = 'document:ready',
+
+    /**
+     * Fired when diagram data changes
+     */
+    documentData = 'document:data',
+    /**
+     * Event type to modify document
+     */
+    documentModify = 'document:modify',
+}
+
+
+/**
+ * Defines the types of postmesssage revice events
+ */
+export enum PostMessageSendEventType{
+    /**
+     * Subscribe to the shape touch
+     */
+    shapeTouchSubscribe = 'shape:touch:subscribe',
+
+
+    /**
+     * Subscribe to selection changes
+     */
+    shapeSelectSubscribe = 'shape:select:subscribe',
+
+    /**
+     * Subscribe to diagram data changes
+     */
+    documentDataSubscribe = 'document:data:subscribe',
+
+    /**
+     * Event type to modify document
+     */
+    documentModify = 'document:modify',
+
+    /**
+     * Event type to set user token
+     */
+    userSetToken = 'user:setToken',
+
+}
+
+export class CreatelyEmbed {
+    private postMessage:PostMessageAPI =  new PostMessageAPI();
+    private documentReady = new BehaviorSubject(false);
+    private documentLoad = new BehaviorSubject(false);
+
+    constructor(private doc:Doctype){
+        const event = PostMessageSendEventType.userSetToken;
+        const message =  {
+            token:this.doc.token
+        };
+        const iframe = document.createElement('iframe');
```

Comment:
Better move this part ( iframe creation and appending it ) to a new function
Also throw an error if no element found for the given Id